python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -760,12 +760,30 @@ def do_evaluate(
             # TODO: ignores ambiguous
             format, strict, _, _ = self.options
             col = self.children[0].evaluate(df, context=context)
+            plc_col = col.obj
+            if format is None:
+                # Polars begins inference with the first non null value
+                int_range = plc.column.Column.from_iterable_of_py(range(col.size))
+                boolmask = plc.unary.is_valid(plc_col)
+
+                table = plc.stream_compaction.apply_boolean_mask(
+                    plc.Table([int_range]), boolmask
+                )
```

Comment:
Not sure if I'm missing anything, but couldn't we apply the boolean mask to the original column instead of a separate indexer?

```python
filtered = plc.stream_compaction.apply_boolean_mask(plc.Table([plc_col]), boolmask).columns()[0]
first_valid_data = plc.copying.get_element(filtered, 0).to_py()
```
?